### ISSUE TITLE
`proper-names` (`MD044`) - false positive in bare URLs

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#734]
### rumdl version

0.2.32

### Installation method

Other

### Bug description

names specified in the `MD044.names` are incorrectly enforced in URLs

### Steps to reproduce

1. create a `.md` file with the following content:
    ```md
    https://foo.com
    ```
2. run `rumdl check`

### Expected behavior

no error

### Actual behavior

```
asdf.md:1:9: [MD044] Proper name 'foo' should be 'Foo' [*]
```

### Configuration

```toml
[global]
enable = ["MD044"]

[MD044]
names = ["Foo"]
```

### Operating system

Linux

### Additional context

looks like this was raised before in #564, but it's still broken on bare URLs.

`no-bare-urls` is disabled in my project because they are automatically hyperlinked by both gitlab and a mkdocs plugin

**Comment by rvben** [comment]
Fixed in v0.2.33 (5e9b51a9). MD044 already knew to skip link destinations and autolinks, but a bare URL isn't a link as far as the parser is concerned, so anything inside one slipped through. It now skips bare URLs and emails too. Good catch @DetachHead, this one mattered: the autofix would happily rewrite foo.com to Foo.com and break the link.
